dcrtimed/backend/filesystem/filesystem.go

```diff
@@ -748,6 +749,81 @@ func (fs *FileSystem) GetTimestamps(timestamps []int64) ([]backend.TimestampResu
 	return gtmes, nil
 }
 
+// Get the last n digests in the added to the Backend
+func (fs *FileSystem) LastDigests(n int32) ([]backend.GetResult, error) {
```

Comment:
This data is saved to the local filesystem, not a database that handles concurrency for you, which means we have to control concurrent access to prevent race conditions.  You need to be holding the read lock for the execution of this function.